sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashOuterJoin.scala

```diff
@@ -59,6 +59,31 @@ override def outputPartitioning: Partitioning = joinType match {
     }
   }
 
+  protected[this] lazy val (buildPlan, streamedPlan) = joinType match {
+    case RightOuter => (left, right)
+    case LeftOuter => (right, left)
+    case x =>
+      throw new IllegalArgumentException(
+        s"BroadcastHashOuterJoin should not take $x as the JoinType")
```

Comment:
Since this code is now in `HashOuterJoin` instead of `BroadcastHashOuterJoin`, I think we should update this error message to reference the new class.